python/cudf/cudf/core/index.py

```diff
@@ -1808,6 +1965,33 @@ def categories(self):
         """
         return self._values.cat().categories
 
+    @annotate("CATEGORICAL_INDEX_EQUALS", color="green", domain="cudf_python")
+    def equals(self, other):
+        """
+        Determine if two Index objects contain the same elements.
+
+        Returns
+        -------
+        out: bool
+            True if “other” is an Index and it has the same elements
+            as calling index; False otherwise.
+        """
+        basic_equality = _check_basic_index_equality(self, other)
+
+        if basic_equality is not None:
+            return basic_equality
+        else:
+            casted_other = other
+            if not is_categorical_dtype(other.dtype):
+                casted_other = other.astype("category")
+                casted_other = casted_other.astype(self.dtype)
```

Comment:
Why are we typecasting to `category` and then to `self.dtype`? Can't we just go straight to `self.dtype`?

Reply:
This will go away with: https://github.com/rapidsai/cudf/pull/5619